File: public/dsa/arrays/primes_upto_n/naive/primes_upto_n.cpp

#include <bits/stdc++.h>
using namespace std;

bool isPrime(int n) {
    if (n < 2) {
        return false;
    }
    for (int i = 2; i * i <= n; i++) {
        if (n % i == 0) {
            return false;
        }
    }
    return true;
}
vector<int> primesUptoN(int n) {
    vector<int> answer;
    for (int i = 2; i <= n; i++) {
        if (isPrime(i) == true) {
            answer.push_back(i);
        }
    }
    return answer;
}

// tests-start
int main() {
    int n = 20;
    auto answer = primesUptoN(n);
    for (int v : answer) {
        cout << v << " ";
    }
    cout << endl;
    return 0;
}
// tests-end

/*output
2 3 5 7 11 13 17 19
*/

File: public/dsa/arrays/primes_upto_n/naive/primes_upto_n.py
from typing import List


class Solution:
    def isPrime(self, n: int) -> bool:
        if n < 2:
            return False
        i = 2
        while i * i <= n:
            if n % i == 0:
                return False
            i += 1
        return True

    def primesUptoN(self, n: int) -> List[int]:
        answer = []
        for i in range(2, n + 1):
            if self.isPrime(i):
                answer.append(i)
        return answer


# tests-start
n = 20
sol = Solution()
answer = sol.primesUptoN(n)
print(answer)
# tests-end

'''output
[2, 3, 5, 7, 11, 13, 17, 19]
'''
